chainntnfs/btcdnotify/btcd.go

```diff
@@ -223,6 +223,21 @@ func (b *BtcdNotifier) startNotifier() error {
 		return err
 	}
 
+	// Before we fetch the best block/blockheight we need to register the
+	// notifications for connected blocks, otherwise we might think we are
+	// at an earlier blockheight because during block notification we
+	// might have already mined some new blocks. Hence we will not get
+	// notified accordingly.
+	//
+	// NOTE: This is only a problem in itests where the block generation
```

Comment:
I don't think it only happens in itests - there could be an edge case where a new block is mined on mainnet and we miss it between rpc calls.

Reply:
removed the comment for itest specifically.